### DISCUSSION TITLE
External execution of code using Tabular Editor

### DISCUSSION THREAD

**Discussion by bernatagulloesbrina** [TabularEditor/TabularEditor3#1346] [🙋♀️TE3 Q&A]
Hello, 

I have a setup on visual studio that allows me to extract the code the way tabular editor wants to execute it, so I can just paste into tabular editor and click play. 

I was wondering if it would be possible to somehow externally execute the code, as-if it was running indeed inside tabular editor, that is, modifiying the data model it is connected to. Maybe saving the code as a file and executing some command-line command for example. 

Is something like this even possible in theory? 

regards!

**Comment by MariaJoseFF** [comment]
Hi @bernatagulloesbrina,

The Command-Line Interface (CLI) can only be used in Tabular Editor 2 for now. A standalone CLI application to accompany Tabular Editor 3 is planned for later release.

You can find documentation about the [Command Line in TE2](https://docs.tabulareditor.com/te2/Command-line-Options.html), which includes a section with an example for [automating script changes](https://docs.tabulareditor.com/te2/Command-line-Options.html#automating-script-changes). It's important to note that the scripts in TE2 can have some limitations, such as not being able to include additional libraries or writing your own methods.

It is important for us to understand our users' needs in order to improve TE3. Could you share some more information on why it would be helpful for you to have the CLI in TE3?

**Comment by bernatagulloesbrina** [comment]
Hello Maria Jose, 

it's not really a hard need right now, but since I try to keep C# development inside visual studio, I was wondering if there would be some way of running the code without copying and pasting into Tabular editor each time . 

At best it would reuse the connection that tabular editor is using, which is probably not great from a security point of view

**Comment by MariaJoseFF** [comment]
Thanks for your feedback, @bernatagulloesbrina. We'll consider your input when we implement the CLI feature.
